spec/generatespec.go

```diff
@@ -0,0 +1,421 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"go/ast"
+	"go/importer"
+	"go/parser"
+	"go/token"
+	"go/types"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"regexp"
+	"strings"
+	"unicode"
+)
+
+// filter is used to filter out go test files.
+func filter(fi os.FileInfo) bool {
+	return !strings.Contains(fi.Name(), "test")
+}
+
+func isExported(name string) bool {
```

Comment:
There is already such a function in `go/ast`: [https://golang.org/pkg/go/ast/#IsExported](https://golang.org/pkg/go/ast/#IsExported).